### PR TITLE
Update docs for 1.23.1

### PR DESCRIPTION
**Pull request by cortinico**
As our website is generated from `main`, I have to push the updates for the docs for `1.23.1` here.

I've generated a new version of the website for `1.23.1`. If we wish we could delete `1.23.0` or keep both.

### CODE REVIEW

File: website/versioned_docs/version-1.23.1/rules/formatting.md [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+
+#### Configuration options:
+
+* ``indentSize`` (default: ``4``)
+
+  indentation size
+
+### AnnotationSpacing
+
+See [ktlint docs](https://pinterest.github.io/ktlint/rules/standard/#annotation-spacing) for documentation.
```
Comment (marschwar) [inline website/versioned_docs/version-1.23.1/rules/formatting.md:33]: Oh no! Not again. The ktlint links are broken again.

Comment (cortinico) [reply website/versioned_docs/version-1.23.1/rules/formatting.md:33]: Oh damn :( What would be the correct url?

Comment (schalkms) [reply website/versioned_docs/version-1.23.1/rules/formatting.md:33]: It seems like ktlint also uses versioned docs.
https://pinterest.github.io/ktlint/0.50.0/

Comment (marschwar) [reply website/versioned_docs/version-1.23.1/rules/formatting.md:33]: See #6099

The generation should be fixed but I don't know if the change can be used to generate docs for 1.x.

**Comment by BraisGabin** [comment]
I would keep both

File: website/versioned_docs/version-1.23.1/rules/formatting.md (hunk 1 above)
Comment (cortinico) [reply website/versioned_docs/version-1.23.1/rules/formatting.md:33]: Ok then I believe we need to manually update this version and pick https://github.com/detekt/detekt/pull/6103 on the release branch for 1.x for the future

Comment (marschwar) [reply website/versioned_docs/version-1.23.1/rules/formatting.md:33]: We could overwrite with the 1.23.0 version from main. There we already updated everything manually

Comment (cortinico) [reply website/versioned_docs/version-1.23.1/rules/formatting.md]: > If not, we can just copy from 1.23.0 and find&replace 0.49.1 -> 0.50.0.

Nope they will not, I've update all the links, could you check them @marschwar ?

Comment (marschwar) [reply website/versioned_docs/version-1.23.1/rules/formatting.md]: Thank you. I checked all the links. All looks good. :fireworks:

**Review by marschwar — APPROVED** [review]

**Pull request merged**